index.bs

```diff
@@ -5294,38 +5294,39 @@ partial interface MLGraphBuilder {
     // of the input values itself, in order to increase the numerical stability of
     // the result.
     // [1]: https://cs231n.github.io/linear-classify/#softmax
-    const max_x = builder.reduceMax(x, { axes: [1], keepDimensions: true });
-    const exp_x = builder.exp(builder.sub(x, max_x));
-    return builder.div(exp_x, builder.reduceSum(exp_x, { axes: [1], keepDimensions: true }));
+    const maxX = builder.reduceMax(x, { axes: [axis], keepDimensions: true });
+    const expX = builder.exp(builder.sub(x, maxX));
+    return builder.div(expX, builder.reduceSum(expX, { axes: [axis], keepDimensions: true }));
   </pre>
 </details>
 </div>
 
-#### {{MLGraphBuilder/softmax(input)}} #### {#api-mlgraphbuilder-softmax-input}
+#### {{MLGraphBuilder/softmax(input, axis)}} #### {#api-mlgraphbuilder-softmax-input-axis}
 <div>
     **Arguments:**
         - *input*: an {{MLOperand}}. The input 2-D tensor.
+        - *axis*: an {{unsigned long}} scalar. The dimension the reduction will be performed on.
 
     **Returns:**
         - an {{MLOperand}}. The output 2-D tensor that contains the softmax results, of the same shape as *input*.
 </div>
 
 <details open algorithm>
   <summary>
-    The <dfn method for=MLGraphBuilder>softmax(|input|)</dfn> method steps are:
+    The <dfn method for=MLGraphBuilder>softmax(|input|, |axis|)</dfn> method steps are:
   </summary>
     1. If [=MLGraphBuilder/validating operand=] with [=this=] and |input| returns false, then [=exception/throw=] a {{TypeError}}.
-    1. If |input|'s [=MLOperand/rank=] is not 2, then [=exception/throw=] a {{TypeError}}.
```

Comment:
Thanks for remove the 2-D restriction in the steps. I think the other "2-D tensor" need to be replaced by "N-D tensor" for the software text, including

> Compute the softmax values of the 2-D input tensor along axis 1.

The axis "1" also needs to be removed.

> input: an MLOperand. The input 2-D tensor.

> Returns: an MLOperand. The output 2-D tensor that contains the softmax results, of the same shape as input.